src/main/java/com/commercetools/sync/services/impl/BaseService.java

```diff
@@ -59,6 +62,12 @@
 
   private static final int MAXIMUM_ALLOWED_UPDATE_ACTIONS = 500;
   static final String CREATE_FAILED = "Failed to create draft with key: '%s'. Reason: %s";
+  /*
+   * A key is a 41 characters long hashed string. (i.e: c25a67e262265fbc36edb33ed0375263586be278) We
```

Comment:
I am really not sure, What is the key size here. So I am asking your suggestions in the PR comment.

Reply:
Since it's not clear in size, you can either count it as max length 256 chars. This will be safe.

As to how to calculate the exact length, check this: https://github.com/sphereio/sphere-product-import/blob/841957349815683d8825855fccaca07a680b894f/lib/common-utils.coffee#L25. As Roman suggested, it might not include headers, so you need to leave space for headers too.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/688/commits/6c5a1d96da9ff116e189543315e418cfcca67813